cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -731,6 +733,11 @@ getPkgConfigDb verbosity progdb = do
     liftIO $ readPkgConfigDb verbosity progdb
 
 
+getDirectoryContentsMonitored :: FilePath -> Rebuild [FilePath]
+getDirectoryContentsMonitored dir = do
+    monitorFiles [monitorDirectory dir]
```

Comment:
This is wrong, you need to monitor the directory contents, not the directory itself.